Add a variable for portsdir, so we can use this variable for building
packages from the FreeBSD ports tree. Also remove the manifest and
mktree files.

--- a/scripts/makepkg.sh.in	2012-04-07 17:03:48.000000000 +0100
+++ b/scripts/makepkg.sh.in.new	2012-12-15 06:36:27.000000000 +0000
@@ -1268,6 +1268,10 @@
 		fi
 	done
 
+	msg2 "$(gettext "Remove +MANIFEST and +MTREE_DIRS")"
+	
+	[[ -f $pkgdir/+MANIFEST ]] && rm -f "$pkgdir"/{+MANIFEST,+MTREE_DIRS}
+
 	# tar it up
 	msg2 "$(gettext "Compressing package...")"
 
@@ -2048,6 +2052,7 @@
 fi
 srcdir="$BUILDDIR/src"
 pkgdir="$BUILDDIR/pkg"
+portsdir="/usr/ports"
 
 PKGDEST=${_PKGDEST:-$PKGDEST}
 PKGDEST=${PKGDEST:-$startdir} #default to $startdir if undefined



There is an issue, when adding packaging to the database about in interrupted
system call, I beleive this to be a timing issue in FreeBSD. However adding
the sleep function fixes this issue.

--- a/scripts/repo-add.sh.in	2012-12-15 06:37:29.000000000 +0000
+++ b/scripts/repo-add.sh.in.new	2012-12-15 06:37:43.000000000 +0000
@@ -259,6 +259,7 @@
 	local line var val
 	while read -r line; do
 		[[ ${line:0:1} = '#' ]] && continue
+		sleep 1
 		IFS=' =' read -r var val < <(printf '%s\n' "$line")
 
 		# normalize whitespace with an extglob

--- a/scripts/makepkg.sh.in	2012-04-07 17:03:48.000000000 +0100
+++ b/scripts/makepkg.sh.in	2012-12-18 02:40:48.000000000 +0000
@@ -1258,6 +1258,7 @@
 
 	# check for changelog/install files
 	for i in 'changelog/.CHANGELOG' 'install/.INSTALL'; do
+		sleep 1
 		IFS='/' read -r orig dest < <(printf '%s\n' "$i")
 
 		if [[ -n ${!orig} ]]; then

